src/YearMonth.php

```diff
@@ -26,16 +29,12 @@ final class YearMonth implements JsonSerializable, Stringable
      */
     private readonly int $year;
 
-    /**
-     * The month, from 1 to 12.
-     */
-    private readonly int $month;
+    private readonly Month $month;
```

Comment:
Should we store the `Month` or its value? We have a precedent with `LocalDate`, that [stores the `int` value and not the enum](https://github.com/brick/date-time/blob/0.6.0/src/LocalDate.php#L59). Not saying this is the right thing to do, but I think we should do the same here for consistency.

Another reason is backwards compatibility: while I'm generally against using it this way, I know that some people compare brick/date-time objects with `==` (or `assertEquals()` in PHPUnit), so this would break here if we're releasing this in a patch release.

We can definitely do this in the next version though (`0.7`), so let's discuss this in another PR/issue if you wish?

Reply:
Sure, didn't thought of the == here. There should be a unit test covering that!

I'll revert it whenever I have some time to do so. And maybe add the test. ;)

Reply:
Done!